### PR TITLE
Remove duplicated "parameter" in E0089 text

### LINKED ISSUE #39732: Error message for mismatched type parameter count repeats "parameter(s)" twice
In Rust 1.15.1 and nightly (24a70eb59 2017-02-09)

```rust
fn foo<A, B>() {}

fn main() {
    foo::<u8>()
}
```

Produces the error:

```
error[E0089]: too few type parameters provided: expected 2 parameters parameters, found 1 parameter parameter
 --> src/main.rs:4:5
  |
4 |     foo::<u8>()
  |     ^^^^^^^^^ expected 2 type parameters
```

Specifically, my issue is with:

> expected 2 **parameters parameters**, found 1 **parameter parameter**

### PR DESCRIPTION
**Pull request by shepmaster**
Closes #39732

### CODE REVIEW

File: src/test/compile-fail/E0089.rs [hunk 1]
```diff
@@ -11,7 +11,6 @@
 fn foo<T, U>() {}
 
 fn main() {
-    foo::<f64>();
-//~^ ERROR E0089
-//~| NOTE expected 2 type parameters
+    foo::<f64>(); //~ ERROR expected 2 parameters, found 1 parameter [E0089]
```
Comment (GuillaumeGomez) [inline src/test/compile-fail/E0089.rs]: This sentence is invalid actually... It needs to speak about type parameters and not just parameter.

Comment (shepmaster) [inline src/test/compile-fail/E0089.rs]: @GuillaumeGomez it does actually, it's just not repeating "type parameter" throughout the *entire* message:

> error[E0089]: too few **type parameters** provided: expected 2 parameters parameters, found 1 parameter parameter

(emphasis mine)

Comment (GuillaumeGomez) [inline src/test/compile-fail/E0089.rs]: Yes, but repeating things is a good thing from my point of view. It allows to have coherent error messages.

**Comment by shepmaster** [comment]
@GuillaumeGomez I've added a second commit that adds the requested "type" or "lifetime" text. Since I was making a larger change, I went ahead and normalized the 4 related errors and their tests.

**Comment by GuillaumeGomez** [comment]
Awesome, thanks! Once the tidy issue is fixed:

```
/checkout/src/test/compile-fail/E0088.rs:15: line longer than 100 chars
/checkout/src/test/compile-fail/E0088.rs:18: line longer than 100 chars
```

I'll r+ it.

**Comment by shepmaster** [comment]
@GuillaumeGomez I always fail to check tidy. Fixed now.

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**